tests/AppSwitcher.test.js

```diff
@@ -0,0 +1,234 @@
+          activeApp="Planner"
+          size={size}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const neetoUIClassPrefix = "neeto-ui-app-switcher__wrapper-size--";
+    expect(
+      document.querySelector(`.${neetoUIClassPrefix}${size}`)
+    ).toBeInTheDocument();
+  });
+
+  it("should render className correctly", () => {
+    const size = "lg";
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          className="test-class"
+          size={size}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    expect(document.querySelector(".test-class")).toBeInTheDocument();
+  });
+
+  it("should close when the back button is pressed", () => {
+    const onClose = jest.fn();
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          onClose={onClose}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const backButton = document.querySelector(
+      "button[data-cy='app-switcher-back-button']"
+    );
+    userEvent.click(backButton);
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  it("should close on pressing escape key", () => {
+    const onClose = jest.fn();
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          closeOnEsc
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          onClose={onClose}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    userEvent.keyboard("{Escape}");
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  it("should close on clicking outside", () => {
+    const onClose = jest.fn();
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          closeOnOutsideClick
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          onClose={onClose}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const appSwitcherBackdrop = document.querySelector(
+      ".neeto-ui-app-switcher__backdrop"
+    );
+    userEvent.click(appSwitcherBackdrop);
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  it("should link subdomains based on environment and subdomain props", () => {
+    const environment = "production";
+    const appName = "neetodesk";
+    const subdomain = "example";
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          subdomain={subdomain}
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          environment={environment}
+        />
+      </AppSwitcherTest>
+    );
+
+    const productionLink = `https://${subdomain}.${appName}.com`;
+    expect(
+      document.querySelector(`a[href='${productionLink}']`)
+    ).toBeInTheDocument();
```

Comment:
Do not use `document` for querying inside tests.

You can either use `screen` which is imported from `@testing-library/react` or the helper functions returned from `render()` function. Prefer `screen` as it is easy to use different types of queries as we need without destructuring the return value of `render()`

This is a good read on common mistakes while writing tests with react-testing-library: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library

I think the above query can be written as below, let me know if it works
```suggestion
    expect(
      screen.getByRole('link', { name: appName })
    ).toBeInTheDocument();
```